Add easing customization to Scroll() screen action

DevilSpiderAkumo opened the pull request:
This pull request adds an optional `easing` argument into the `Scroll()` screen action.
The current implementation of this action only allows the `ease` warper, so I thought it would be nice to have more easing options.

DevilSpiderAkumo committed:
Update 00action_other.rpy

Scroll() improvements to allow changing of scroll easing.

DevilSpiderAkumo commented:
Oh, true. Let me just do another commit with this in mind.

mal commented:
We have existing public APIs allowing warpers to be used already, such as in `ImageDissolve`, `MoveTransition` and `Drag`, i.e:
```rpy
define circirisout = ImageDissolve("circiris.png", 1.0, time_warp=_warper.easeout)
```
They all expect to be provided the warper function, not a string. I think for consistency it would be better to match that approach here, and also give it a similar name such that the default would be `warper=_warper.ease`. This would also give the flexibility present in those other APIs of being able (and indeed expected) to provide a function - be it inbuilt or user defined.

Andykl commented:
I agree with mal that consistency is more important here. It feels weird that you need to reach 'private' _wraper' module for common ones, so maybe later we will fix all instances to take name or function, or to alias _warper to something different, but it would be separate issue.

DevilSpiderAkumo committed:
Change the "easing" kwarg in Scroll() to use "time_warp"

This was made in order to ensure consistency with other APIs that support easing.

DevilSpiderAkumo committed:
Merge branch 'renpy:master' into master
